src/test_workflow/benchmark_test/benchmark_test_cluster.py

```diff
@@ -0,0 +1,158 @@
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+
+import json
+import logging
+import os
+import subprocess
+from contextlib import contextmanager
+from typing import Any, Generator
+
+import requests
+from requests.auth import HTTPBasicAuth
+from retry.api import retry_call  # type: ignore
+
+from manifests.bundle_manifest import BundleManifest
+from test_workflow.benchmark_test.benchmark_args import BenchmarkArgs
+
+
+class BenchmarkTestCluster:
+    manifest: BundleManifest
+    work_dir: str
+    current_workspace: str
+    args: BenchmarkArgs
+    output_file: str
+    params: str
+    is_endpoint_public: bool
+    cluster_endpoint: str
+    cluster_endpoint_with_port: str
+
+    """
+    Represents a performance test cluster. This class deploys the opensearch bundle with CDK. Supports both single
+    and multi-node clusters
+    """
+
+    def __init__(
+            self,
+            bundle_manifest: BundleManifest,
+            config: dict,
+            args: BenchmarkArgs,
+            current_workspace: str
+    ) -> None:
+        self.manifest = bundle_manifest
+        self.current_workspace = current_workspace
+        self.args = args
+        self.output_file = "output.json"
+        role = config["Constants"]["Role"]
+        params_dict = self.setup_cdk_params(config)
+        params_list = []
+        for key, value in params_dict.items():
+            print(key, value)
+            if value:
+                if key == 'additionalConfig':
```

Comment:
Why is this needed?

Reply:
It has more to do how opensearch-cluster-cdk accepts the json type parameter, the input has to be enclosed in single-quote in order to be properly parsed in the CDK Code [here](https://github.com/opensearch-project/opensearch-cluster-cdk/blob/main/lib/os-cluster-entrypoint.ts#L139-L147). 
So in the above code I am just making sure that if the parameter is of type `additional-config` make sure the value is enclosed in single-quotes. Hope that makes sense. 
@dblock

Reply:
It seems weird. Is there a more generic version of this? For example if the value has quotes it should be single quoted?

Reply:
Does quoting everything work, ie. doing `params_list.append(f" -c {key}=\'{value}\'")` all the time?

If not, add a TODO: fixme here or something like that with an explanation that this is a hack

Next, if you want to clean this up further,  you can write the loop with a `map` and a lambda that will make this a lot shorter. Finally, add role params that are `-c` into `params_dict` and let them be transformed with this code to avoid hard-coding strings. You'll still need the `--require-approval` and the `--plugin` params I guess. I'm ok to leave it as is, but consider doing it.

Reply:
Ack! Will add a TODO and take this up in the next iteration.